ocaml/xapi-idl/lib/xcp_service.ml

```diff
@@ -364,7 +364,12 @@ let canonicalise x =
   if not (Filename.is_relative x) then
     x
   else (* Search the PATH and XCP_PATH for the executable *)
-    let paths = split_c ':' (Sys.getenv "PATH") in
+    let paths =
```

Comment:
This looks too complicated but it was like this before. We are looking for the first hit searching along the directories in PATH. This could be expressed using `List.find_opt` over the elements of PATH.